### PR TITLE
Migrate to null safety

### LINKED ISSUE #63: Support Null Safety
I am trying to upgrade [Exercism/Dart](https://github.com/exercism/dart) to null safety

But it seems that io (as well as others) has not yet migrated to support nullsafety. Is there a timeline on when this could be added?

No strict deadline on my part, just creating as a way to track this.

The other dependency is dart_style and that is being tracked here: https://github.com/dart-lang/dart_style/issues/978

### PR DESCRIPTION
**Pull request by simolus3**
The `wrap` api is kind of awkward to use with null-safety, it might be easier to just disallow `null` inputs? I kept compatibility to existing apis.

Closes #63.

### CODE REVIEW

**Comment by kevmoo** [comment]
@simolus3 – please rebase this on HEAD!

**Comment by simolus3** [comment]
Thanks for the heads-up (pun intended). I rebased this on master and fixed new analysis warnings that happened during the migration.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.4.0
+
+- Migrate this package to null-safety
+
 ## 0.3.6-dev
```
Comment (kevmoo) [inline CHANGELOG.md:5]: Delete the `0.3.6-dev` header here. It'll all roll up to v0.4.0

File: pubspec.yaml [hunk 2]
```diff
@@ -2,19 +2,24 @@ name: io
...
+  string_scanner: ^1.1.0
 
 dev_dependencies:
   dart_style: ^1.0.7
-  pedantic: ^1.2.0
-  test: ^1.0.0
-  test_descriptor: ^2.0.0
+  pedantic: ^1.10.0
+  test: ^1.16.0
+  test_descriptor: ^2.0.0-nullsafety
```
Comment (kevmoo) [inline pubspec.yaml:20]: remove the `-nullsafety` bit here. This is published!

**Review by kevmoo — APPROVED** [review]
Thanks so much!

**Pull request merged**